File: src/services/ai.ts
import { invoke } from "@tauri-apps/api/core";

export interface AIConfig {
  apiKey: string;
  model: string;
  maxTokens: number;
}

export async function queryAI(query: string, config: AIConfig): Promise<string> {
  try {
    const result = await invoke<string>("query_ai", {
      query,
      config: {
        api_key: config.apiKey,
        model: config.model,
        max_tokens: config.maxTokens,
      },
    });
    return result;
  } catch (error) {
    console.error("AI query failed:", error);
    throw error;
  }
}

export async function testAIConnection(config: AIConfig): Promise<string> {
  try {
    const result = await invoke<string>("test_ai_connection", {
      config: {
        api_key: config.apiKey,
        model: config.model,
        max_tokens: config.maxTokens,
      },
    });
    return result;
  } catch (error) {
    console.error("AI connection test failed:", error);
    throw error;
  }
}

File: src-tauri/src/commands/ai.rs
use crate::ai::query_openrouter;
use crate::ai::types::AIConfig;

#[tauri::command]
pub async fn query_ai(query: String, config: AIConfig) -> Result<String, String> {
    query_openrouter(&config, &query)
        .await
        .map_err(|e| e.to_string())
}

#[tauri::command]
pub async fn test_ai_connection(config: AIConfig) -> Result<String, String> {
    let result = query_ai("What is 2+2?".to_string(), config).await?;
    Ok(format!("Connected! Response: {}", result))
}
